notebooks/classify/signature.py

```diff
@@ -0,0 +1,249 @@
+import numpy as np
+import pandas as pd
+import re
+import datetime
+import matplotlib.pyplot as plt
+import ipywidgets as widgets
+from typing import Dict, List
+
+class DataSignature:
+    def __init__(self, table: pd.DataFrame, DEBUG: bool = False):
```

Comment:
If you separate internal methods from public methods it's going to be clearer what you want the user to do once they've created the class.
Ideally you'd want something like this:
data_sig = DataSignature(dframe)
data_sig.generate_signatures()

You probably don't need any additional parameters (I don't think) - You might want to output some feedback from the main function if it's going to take a while to complete.